crates/oxc_linter/src/rules/eslint/max_params.rs

```diff
@@ -0,0 +1,143 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{CompactStr, Span};
+use serde_json::Value;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(max-params): {0:?}")]
+#[diagnostic(
+    severity(warning),
+    help("This rule enforces a maximum number of parameters allowed in function definitions.")
+)]
+struct MaxParamsDiagnostic(CompactStr, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct MaxParams(Box<MaxParamsConfig>);
+
+#[derive(Debug, Clone)]
+pub struct MaxParamsConfig {
+    max: usize,
+}
+
+impl std::ops::Deref for MaxParams {
+    type Target = MaxParamsConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for MaxParamsConfig {
+    fn default() -> Self {
+        Self { max: 3 }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforce a maximum number of parameters in function definitions
+    ///
+    /// ### Why is this bad?
+    /// Functions that take numerous parameters can be difficult to read and write because it requires the memorization of what each parameter is, its type, and the order they should appear in. As a result, many coders adhere to a convention that caps the number of parameters a function can take.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// function foo (bar, baz, qux, qxx) {
+    ///     doSomething();
+    /// }
+    /// ```
+    MaxParams,
+    style
+);
+
+impl Rule for MaxParams {
+    fn from_configuration(value: Value) -> Self {
+        let config = value.get(0);
+        if let Some(max) = config
+            .and_then(Value::as_number)
+            .and_then(serde_json::Number::as_u64)
+            .and_then(|v| usize::try_from(v).ok())
+        {
+            Self(Box::new(MaxParamsConfig { max }))
+        } else {
+            let max = config
+                .and_then(|config| config.get("max"))
+                .and_then(Value::as_number)
+                .and_then(serde_json::Number::as_u64)
+                .map_or(3, |v| usize::try_from(v).unwrap_or(3));
+
+            Self(Box::new(MaxParamsConfig { max }))
+        }
+    }
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::Function(function) => {
```

Comment:
You can get the function name by `function.id`

Reply:
Mmm, ok, and for ArrowFunction? I think it doesn't have id property

Reply:
Feels like we can ignore the name for `ArrowFunction`? [ESLint](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG1heC1wYXJhbXM6IFtcImVycm9yXCIsIDNdKi9cbi8qZXNsaW50LWVudiBlczYqL1xuXG5mdW5jdGlvbiBmb28xIChiYXIsIGJheiwgcXV4LCBxeHgpIHtcbiAgICBkb1NvbWV0aGluZygpO1xufVxuXG5sZXQgZm9vMiA9IChiYXIsIGJheiwgcXV4LCBxeHgpID0+IHtcbiAgICBkb1NvbWV0aGluZygpO1xufTsifQ==) also omits it.

Reply:
wow, I didn't know this tool!, I have already applied the change